Editor/AGS.Editor/Panes/FreePanControl.cs

```diff
@@ -0,0 +1,98 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Text;
+using System.Windows.Forms;
+
+namespace AGS.Editor
+{
+    public class FreePanControl : BufferedPanel
+    {
+        private const int WS_EX_COMPOSITED = 0x02000000;
+        private int _lastX, _lastY; //used to prevent useless mouse move events
+        private Point _panGrabPoint = Point.Empty;
+        private Cursor _panCursor = Cursors.Hand; // We need a better hand cursor, using the "link" hand for now
+        public event EventHandler PanGrabbed;
+        public event EventHandler PanReleased;
+
+        // Enabling Composite makes scrolling smoother
+        protected override CreateParams CreateParams
+        {
+            get
+            {
+                CreateParams cp = base.CreateParams;
+                cp.ExStyle |= WS_EX_COMPOSITED;
+                return cp;
+            }
+        }
+
+        public FreePanControl() : base()
+        {
+
+        }
+
+        protected virtual void OnPanGrab(EventArgs e)
+        {
+            PanGrabbed?.Invoke(this, EventArgs.Empty);
+        }
+
+        protected virtual void OnPanRelease(EventArgs e)
+        {
+            PanReleased?.Invoke(this, EventArgs.Empty);
+        }
+
+        protected override void OnMouseDown(MouseEventArgs e)
+        {
+            // defines the shortcut for click pan Control+Shift+Left Click Hold or Middle Click Hold without any modifier.
+            if ( (e.Button == MouseButtons.Left   && Control.ModifierKeys == (Keys.Control|Keys.Shift)) ||
+                 (e.Button == MouseButtons.Middle && Control.ModifierKeys == Keys.None) )
+            {
+                OnPanGrab(e);
+                _panGrabPoint = e.Location;
+                Cursor = _panCursor;
+                return;
```

Comment:
Is the `return` intentional? It's normally anti-pattern to not continue to `base` when overriding `virtual` functions, in this case someone who might have subscribed to the `MouseDown` event will be expecting their event to trigger, but it won't in certain cases because you circumvented it.

Reply:
the idea is to prevent firing the events like editing the walkable areas. The alternative way is adding a property like `IsGrabbing` and only process the Events in the Room Editor if `IsGrabbing` is false. My mentality came from using `ClaimEvent` in AGS Script.

Reply:
I moved to using IsGrabbing and placed the Return in the code. There's only one return now in Event handling portion but it's actually solving an unexpected behavior from WinForms, so I will leave there.